examples/tutorials/advanced/grid_equalization.py

```diff
@@ -0,0 +1,223 @@
+"""
+Performing grid histogram equalization
+--------------------------------------
+The :meth:`pygmt.grdhisteq.equalize_grid` function creates a grid with
+statistics based on a cumulative distribution function.
+"""
+# sphinx_gallery_thumbnail_number = 3
+
+import pygmt
+
+###############################################################################
+# Load sample data
+# ----------------
+# Load the sample Earth relief data for a region around Yosemite valley
+# and use :meth:`pygmt.grd2xyz` to create a :class:`pandas.Series` with the
+# z values.
+
+grid = pygmt.datasets.load_earth_relief(
+    resolution="03s", region=[-119.825, -119.4, 37.6, 37.825]
+)
+grid_dist = pygmt.grd2xyz(grid=grid, output_type="pandas")["elevation"]
+
+###############################################################################
+# Plot the original digital elevation model and data distribution
+# ---------------------------------------------------------------
+# For comparison, we will create a map of the original digital elevation
+# model and a histogram showing the distribution of data values.
+
+# Create an instance of the Figure class
+fig = pygmt.Figure()
+# Define figure configuration
+pygmt.config(FORMAT_GEO_MAP="ddd.x", MAP_FRAME_TYPE="plain")
+# Define the colormap for the figure
+pygmt.makecpt(series=[500, 3540], cmap="turku")
+# Define the position of the colormap
+cmap_position = "JMR+o1c/0c+w3c/0.3c"
```

Comment:
The variable `cmap_position` is defined here but is also used in other examples. Thus other examples cannot be run directly without defining the variable. It would be more convenient if users can simply copy and run these examples with little efforts (e.g., `import pygmt`).